osquery/core/init.cpp

```diff
@@ -392,12 +460,31 @@ void Initializer::initDaemon() const {
   systemLog(binary_ + " started [version=" + kVersion + "]");
 
   if (!FLAGS_ephemeral) {
-    // Create a process mutex around the daemon.
-    auto pid_status = createPidFile();
-    if (!pid_status.ok()) {
-      LOG(ERROR) << binary_ << " initialize failed: " << pid_status.toString();
+    auto pidfile_path = fs::path(FLAGS_pidfile).make_preferred().string();
+
+    auto pidfile_res = Pidfile::create(pidfile_path);
+    if (pidfile_res.isError() &&
+        pidfile_res.getErrorCode() == Pidfile::Error::Busy && FLAGS_force) {
+      if (terminateActiveOsqueryInstance()) {
+        for (int retry = 0; retry < 5; ++retry) {
+          sleepFor(2000);
+
+          pidfile_res = Pidfile::create(pidfile_path);
+          if (pidfile_res.isValue()) {
+            break;
+          }
+        }
+      }
+    }
```

Comment:
I see that there's a `LOG(ERROR)` down there, but it feels like we can be more explicit here -- we know the terminate failed, should we log it? TBH, maybe log each failure in VLOG?

Reply:
The kill procedure only happens once, and contains a LOG call at each step that can fail. I have added an additional log message at each retry